iothub/device/src/Common/Extensions/CommonExtensions.cs

```diff
@@ -202,5 +202,16 @@ public static int NthIndexOf(this string str, char value, int startIndex, int n)
 
             return entryIndex;
         }
+
+        /// <summary>
+        /// Helper to remove extra whitespace from the supplied string.
```

Comment:
Why is this an important method to call?

Reply:
so that we reduce the no of bytes being sent on the wire